test/extended/openstack/kuryr.go

```diff
@@ -0,0 +1,164 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+var _ = g.Describe("[sig-installer][Feature:openstack][Kuryr] The OpenStack platform", func() {
+	var namespaces []*v1.Namespace
+	var networkClient *gophercloud.ServiceClient
+	var dc dynamic.Interface
+	var clientSet *kubernetes.Clientset
+
+	g.BeforeEach(func() {
+		g.By("preparing openshift dynamic client")
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		c, err := configv1client.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+
+		if network.Status.NetworkType != "Kuryr" {
+			e2eskipper.Skipf("Test can only be run with Kuryr SDN")
+		}
+
+		g.By("preparing openstack client")
+		networkClient, err = client(serviceNetwork)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+	})
+
+	g.AfterEach(func() {
+		clientSet, err := e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By("Cleanup up Namespaces")
+		for _, ns := range namespaces {
+
+			o.Expect(clientSet.CoreV1().Namespaces().Delete(context.TODO(), ns.Name, metav1.DeleteOptions{})).To(o.Succeed())
+		}
+	})
+	g.It("should create a subnet when a pod without hostNetwork is created", func() {
+
+		g.By("Creating a Namespace and pods")
+		{
+
+			clientSet, err := e2e.LoadClientset()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			ns := CreateNamespace(clientSet, "kuryr-hostnetwork")
+
+			namespaces = append(namespaces, ns)
+			listOpts := subnets.ListOpts{
+				Name: "ns/" + ns.Name + "-subnet",
```

Comment:
@gryf: Will this be invalidated with latest Kuryr changes? Can we suggest another way of finding the namespace for @itzikb-redhat?

Reply:
Right. The changes aren't merged yet, but eventually will be. so the name for the subnet would be `ns.Name` instead of `'ns/' + ns.Name + '-subnet'`.

Reply:
I changed the code to reflect it but I don't have a setup with the new changes yet.